### PR TITLE
[feat] Add TabTransformer and AutoGluon example notebooks as Amazon SageMaker algorithms

### PR DESCRIPTION
**Pull request by Neo9061**
1. Added TabTransformer and AutoGluon example notebooks as Amazon SageMaker algorithms.
2. Modified existing JumpStart example notebooks LightGBM, CatBoost (`jumpstart_tabular_classification/Amazon_JumpStart_Tabular_Classification_LightGBM_CatBoost.ipynb`, `jumpstart_tabular_regression/Amazon_JumpStart_Tabular_Regression_LightGBM_CatBoost.ipynb`), XGBoost, Linear Learner (`jumpstart_tabular_classification/Amazon_JumpStart_Tabular_Classification_XGBoost_LinearLearner.ipynb`, and `jumpstart_tabular_regression/Amazon_JumpStart_Tabular_Regression_XGBoost_LinearLearner.ipynb`) and changed the keyword from `JumpStart` to `Amazon SageMaker algorithms`
3. Moved LightGBM, CatBoost example notebooks from previous directory `JumpStart_Tabular_Regression` and `JumpStart_Tabular_Classification` into a new directory `lightgbm_catboost_tabular` to be consistent with the directory structure of all the other Amazon SageMaker algorithms.
4. Similarly, moved XGBoost and Linear Learner example notebooks from `JumpStart_Tabular_Regression` and `JumpStart_Tabular_Classification` into a new directory `xgboost_linear_learner_tabular` to be consistent with the directory structure of all the other Amazon SageMaker algorithms.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [ ] I have tested my notebook(s) and ensured it runs end-to-end
- [ ] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: introduction_to_amazon_algorithms/autogluon_tabular/Amazon_Tabular_Classification_AutoGluon.ipynb [hunk 1]
```diff
@@ -0,0 +1,577 @@
+{
```
Comment (jkroll-aws) [inline introduction_to_amazon_algorithms/autogluon_tabular/Amazon_Tabular_Classification_AutoGluon.ipynb:1]: <p>Branding nit: replace all occurrences of <code>Sagemaker</code> with <code>SageMaker</code>   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3461/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: introduction_to_amazon_algorithms/autogluon_tabular/Amazon_Tabular_Regression_AutoGluon.ipynb [hunk 2]
```diff
@@ -0,0 +1,564 @@
+{
```
Comment (jkroll-aws) [inline introduction_to_amazon_algorithms/autogluon_tabular/Amazon_Tabular_Regression_AutoGluon.ipynb:1]: <p>Style nit: I think there's no need to have <code>ABALONE</code> in all caps -&gt; <code>Abalone</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3461/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

File: introduction_to_amazon_algorithms/autogluon_tabular/Amazon_Tabular_Classification_AutoGluon.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Neo9061) [reply introduction_to_amazon_algorithms/autogluon_tabular/Amazon_Tabular_Classification_AutoGluon.ipynb:1]: <p>I can check but I don't think so. And this is consistent with all the other SDK code for JumpStart models.</p>

Comment (Neo9061) [reply introduction_to_amazon_algorithms/autogluon_tabular/Amazon_Tabular_Classification_AutoGluon.ipynb:1]: <p>Thanks for catching that! I have addressed for all notebooks in upcoming PR.</p>

File: introduction_to_amazon_algorithms/autogluon_tabular/Amazon_Tabular_Regression_AutoGluon.ipynb (hunk 2 above)
Comment (Neo9061) [reply introduction_to_amazon_algorithms/autogluon_tabular/Amazon_Tabular_Regression_AutoGluon.ipynb:1]: <p>Thanks! I have changed them in all regression notebook</p>

**Comment by zhaoqizqwang** [comment]
Closing this PR since it was created a long time ago and didn't pass all the checks. Feel free to open a new pull request instead.

**Pull request closed without merging**